apps/storefront/lib/auth/useAuthChange.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from "react";
+import { SaleorAuthEvent, STORAGE_AUTH_EVENT_KEY } from "./SaleorAuthStorageHandler";
+
+interface UseAuthChangeProps {
+  onSignedIn?: () => void;
+  onSignedOut?: () => void;
+}
+
+// used to handle client cache invalidation on login / logout and when
+// token refreshin fails
+export const useAuthChange = ({ onSignedOut, onSignedIn }: UseAuthChangeProps) => {
```

Comment:
Isn't this the same logic we've added for checkout?
Maybe it should be extracted into a separate package in monorepo?

Reply:
It's literally copy paste, @mmiszy proposed we leave it like that for now. I guess it makes sense since the devtools team might soon want to make a whole separate project out of this.